src/btree/bt_read.c

```diff
@@ -55,13 +55,16 @@ __wt_cache_read(WT_SESSION_IMPL *session, WT_REF *ref)
 		WT_ERR(__wt_btree_new_leaf_page(session, &page));
 		ref->page = page;
 	} else {
-		/* Read the backing disk page. */
+		/*
+		 * Read the page, then build the in-memory version of the page.
+		 * Clear any local reference to an allocated copy of the disk
+		 * image on return, the page steals it.
+		 */
 		WT_ERR(__wt_bt_read(session, &tmp, addr, addr_size));
-
-		/* Build the in-memory version of the page. */
 		WT_ERR(__wt_page_inmem(session, ref, tmp.data,
 		    WT_DATA_IN_ITEM(&tmp) ?
 		    WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED, &page));
+		tmp.mem = NULL;
```

Comment:
Right, but once `__wt_page_inmem` returns, `ref->page` "owns" the memory, and `__wt_ref_out` will discard it.